bevm/service.go

```diff
@@ -13,43 +11,9 @@ func init() {
 	log.AddUserUninterestingGoroutine(
 		"go-ethereum/core.(*txSenderCacher).cache")
 
-	_, err := onet.RegisterNewService(ContractBEvmID, newServiceBEvm)
-	log.ErrFatal(err)
-	_, err = onet.RegisterNewService(ContractBEvmValueID, newServiceBEvmValue)
-	log.ErrFatal(err)
-}
-
-// Service structure for BEvm contracts
-type Service struct {
-	// We need to embed the ServiceProcessor, so that incoming messages
-	// are correctly handled.
-	*onet.ServiceProcessor
-}
-
-func newServiceBEvm(c *onet.Context) (onet.Service, error) {
-	s := &Service{
-		ServiceProcessor: onet.NewServiceProcessor(c),
-	}
-
-	err := byzcoin.RegisterContract(c, ContractBEvmID, contractBEvmFromBytes)
-	if err != nil {
-		return nil, xerrors.Errorf("failed to registe BEvm contract: %v", err)
-	}
-
-	return s, nil
-}
-
-func newServiceBEvmValue(c *onet.Context) (onet.Service, error) {
-	s := &Service{
-		ServiceProcessor: onet.NewServiceProcessor(c),
-	}
-
-	// BEvmValue does not support explicit creation, so we can pass nil
-	err := byzcoin.RegisterContract(c, ContractBEvmValueID, nil)
-	if err != nil {
-		return nil, xerrors.Errorf("failed to register BEvmValue "+
-			"contract: %v", err)
-	}
-
-	return s, nil
+	// Register contracts
+	log.ErrFatal(byzcoin.RegisterGlobalContract(ContractBEvmID,
+		contractBEvmFromBytes))
+	log.ErrFatal(byzcoin.RegisterGlobalContract(ContractBEvmValueID,
```

Comment:
Is this needed at all?

Reply:
I mean the registration of `ContractBEvmValueID`

Reply:
Yes, see my comment on #2206...